File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let overlap = strs[0]
    for (let i = 1; i < strs.length; i++) {
        let word = strs[i]
        if (overlap.length > word.length) {
            for (let i = 0; i < overlap.length; i++) {
                if (overlap[i] === word[i]) {
                    continue
                } else {
                    overlap = overlap.slice(0, i)
                    break
                }
            }
        } else {
            for (let i = 0; i < word.length; i++) {
                if (overlap[i] === word[i]) {
                    continue
                } else {
                    overlap = overlap.slice(0, i)
                    break
                }
            }
            
        }
    }
    return overlap
};

File: 14-longest-common-prefix/14-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
       //strs length is 0, return empty string
        if (strs.length == 0) {
            return "";
        }
        //set prefix as first elem of strs
        String prefix = strs[0];
        
        //iterate from index 1 to end of strs
        for (int i = 1; i < strs.length; i++) {
            String word = strs[i];
            while (word.indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);        
            }
            if (prefix.length() == 0) {
                return "";
            }
            
        }
        // check if each string contains prefix
        //if not
        //while the string does not contain prefix
        //cut off the last character of the prefix
        //if prefix length is ever 0, return empty string
        
        //return the prefix
        return prefix;       
    }    
}
